modules/termination-state-aws/src/main/java/org/opencastproject/terminationstate/endpoint/aws/AutoScalingTerminationStateRestService.java

```diff
@@ -0,0 +1,126 @@
+        notes = {
+        "All paths above are relative to the REST endpoint base (something like http://your.server/termination/aws/autoscaling)",
+        "If the service is down or not working it will return a status 503, this means the the underlying service is "
+                + "not working and is either restarting or has failed",
+        "A status code 500 means a general failure has occurred which is not recoverable and was not anticipated. In "
+                + "other words, there is a bug! You should file an error report with your server logs from the time when the "
+                + "error occurred: <a href=\"https://opencast.jira.com\">Opencast Issue Tracker</a>" })
+public class AutoScalingTerminationStateRestService implements TerminationStateRestService {
+
+  private static final Log logger = new Log(LoggerFactory.getLogger(AutoScalingTerminationStateRestService.class));
+
+  private TerminationStateService service;
+
+  @Override
+  @GET
+  @Path("/state")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RestQuery(name = "stateasjson", description = "Returns the Termination State as JSON.  Possible termination states are none, wait and ready.", returnDescription = "A JSON representation of the termination state.",
+          reponses = {
+            @RestResponse(responseCode = SC_OK, description = "A JSON representation of the termination state."),
+            @RestResponse(responseCode = SC_SERVICE_UNAVAILABLE, description = "The AWS Autoscaling Termination State Service is disabled or unavailable")
+          })
+  public Response getState() {
+    if (service != null) {
+      JSONObject json  = new JSONObject();
+      String state = service.getState().toString();
+      json.put("state", state);
+      return Response.ok(json.toJSONString()).build();
+    } else {
+      logger.error("TerminationStateService is not available");
+      return Response.status(Response.Status.SERVICE_UNAVAILABLE).build();
+    }
+  }
+
+  @Override
+  @PUT
+  @Path("/state")
+  @RestQuery(name = "setstate", description = "Set the termination state. The only permissable value to write to the state is 'wait'", returnDescription = "Whether the termination state was set successfully",
+          restParameters = {
+            @RestParameter(name = "state", type = Type.STRING, defaultValue = "wait", description = "The termination state, the only valid value is 'wait'", isRequired = false)
+          },
+          reponses = {
+            @RestResponse(responseCode = SC_NO_CONTENT, description = "The node is preparing to terminate"),
+            @RestResponse(responseCode = SC_BAD_REQUEST, description = "The state was not 'wait'"),
+            @RestResponse(responseCode = SC_SERVICE_UNAVAILABLE, description = "The AWS Autoscaling Termination State Service is disabled or unavailable"),
+          })
+  public Response setState(@FormParam("state") String state) {
+    if (service != null) {
+      if ("wait".equalsIgnoreCase(state)) {
```

Comment:
Constant should either be a class constant or (ideally) constructed from AWS provided constants.